daliuge-engine/run_engine.sh

```diff
@@ -2,6 +2,7 @@
 DOCKER_OPTS="\
 --shm-size=2g --ipc=shareable \
 --rm \
+--gpus=all \
```

Comment:
as a sanity check, could someone test that this doesn't cause an unknown argument on environment without nvidia-docker?

Reply:
I'll give it a go. This is my first time using these build/run scripts based on Docker so it might take a bit

Reply:
```
Running Engine development version in background...
docker run -td --shm-size=2g --ipc=shareable --rm --gpus=all --name daliuge-engine -v /var/run/docker.sock:/var/run/docker.sock -p 5555:5555 -p 6666:6666 -p 8000:8000 -p 8001:8001 -p 8002:8002 -p 9000:9000 --user 1000:1000  --group-add 134 -v /home/rtobar/dlg/workspace/settings/passwd:/etc/passwd -v /home/rtobar/dlg/workspace/settings/group:/etc/group -v /home/rtobar/dlg:/home/rtobar/dlg --env DLG_ROOT=/home/rtobar/dlg  icrar/daliuge-engine:yan-973
51a0d17da1384a237d5271cf067bc0d3b37cedd0a05d3222f6d699895cda750c
docker: Error response from daemon: could not select device driver "" with capabilities: [[gpu]].
```

Reply:
(and indeed without the `--gpus=all` option it starts)

Reply:
thanks, I think the best workaround in this case may be wrap the arg based on the presence of nvidia-docker, e.g.
```
$([[ $(nvidia-docker version) ]] && echo '--gpus=all' || echo '')
```